### PR TITLE
#73 Add more detailed logging

### PR DESCRIPTION
**Pull request by tomuben**

### CODE REVIEW

File: src/r_exasol/connection/async_executor/async_executor_impl.h [hunk 1]
```diff
@@ -44,18 +49,25 @@ namespace exa {
...
+                ae_debug_printer::print("is joinable!. Call join()");
                 mThread.join();
+                ae_debug_printer::print("Finished join!");
             }
             return mQueryExecutor.getQueryExecutorResult();
         }
 
     private:
         /**
          * Executes the remote execution in a bg thread. If an error occurs it calls errorHandler.
```
Comment (tkilias) [inline src/r_exasol/connection/async_executor/async_executor_impl.h:63]:
```suggestion
         * Executes the remote execution in a background thread. If an error occurs it calls errorHandler.
```

**Review by tkilias — CHANGES_REQUESTED** [review]
Also, check if you can replace the hard-coded function names in the stack loggers with macros for the function name.

File: doc/changes/changes_6.0.0.md [hunk 2]
```diff
@@ -1,19 +1,22 @@
 # r-exasol 6.0.0, released 2021-08-27
 
-Code name: t.b.d.
+Code name: Encryption support.
 
 ## Summary 
 
-t.b.d. 
+This released contains a major refactoring of the C++ layer. This also includes the removal of several C functions, which were exposed to the R-layer, but effectively unused. Also, r-exasol now supports encryption of the data channel via TLS.
```
Comment (tkilias) [inline doc/changes/changes_6.0.0.md:7]:
```suggestion
This release contains a major refactoring of the C++ layer. This also includes the removal of several C functions, which were exposed to the R-layer, but effectively unused. Also, r-exasol now supports encryption of the data channel via TLS.
```

File: doc/changes/changes_6.0.0.md [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 
 ## Features / Enhancements
  #63 refactor c layer
```
Comment (tkilias) [inline doc/changes/changes_6.0.0.md:10]:
```suggestion
 #63 Refactor C layer
```

File: src/r_exasol/connection/async_executor/async_executor_impl.h [hunk 4]
```diff
@@ -24,6 +28,7 @@ namespace exa {
          * @param errorHandler Callback function to invoke if an error occurs.
          */
         void execute(exa::tBackgroundAsyncErrorFunction errorHandler) override {
+            ae_stack_trace_logger astl(__func__);
```
Comment (tkilias) [inline src/r_exasol/connection/async_executor/async_executor_impl.h:31]: Is there a reason why you don't use the macro STACK_PRINTER

Comment (tomuben) [reply src/r_exasol/connection/async_executor/async_executor_impl.h:31]: I didn't like the idea of getting this macro then in all other files, which include this header.

Comment (tkilias) [reply src/r_exasol/connection/async_executor/async_executor_impl.h:31]: ah, I see, the files which use the macro are cpp files and not headers. Yeah, that sounds reasonable.

**Review by tkilias — APPROVED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**